src/block/Trapdoor.php

```diff
@@ -85,10 +87,22 @@ public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Blo
 	}
 
 	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null, array &$returnedItems = []) : bool{
+		$this->toggle();
+
+		return true;
+	}
+
+	public function onProjectileInteraction(Projectile $projectile) : void{
+		if($projectile instanceof WindCharge && $this->getTypeId() !== BlockTypeIds::IRON_TRAPDOOR){
+			$this->toggle();
+		}
+	}
+
+	public function toggle() : void {
```

Comment:
This should be `private` or `protected`, making it public has implications beyond this PR's scope